test/integration/targets/win_disk_facts/tasks/main.yml

```diff
@@ -0,0 +1,13 @@
+# NOTE: The win_disk_facts module only works on Win2012R2+
+
+- name: Test Windows capabilities
+  raw: Get-Command Get-Disk -ErrorAction SilentlyContinue; return $?
+  failed_when: no
+  register: new_diskfacts
+
+- name: Only run tests when Windows is capable
```

Comment:
You should also have a test for a failure when `Get_Disk` is not available to ensure it fails with an error that makes sense.

Reply:
I have implemented a check which is throwing a message now.
The integration test seems to work because it is throwing the message in shippable tests for Win2008R2:

2017-12-15 14:25:26 fatal: [windows-2008-SP2]: FAILED! => {
2017-12-15 14:25:26     "changed": false, 
2017-12-15 14:25:26     "msg": "Cmdlet Get-Disk is not available in your Windows version"
2017-12-15 14:25:26 }

But this also cancels the shippable test with an error.
How can I avoid this?

Reply:
You can add "failed_when:" and define it to only fail when the "msg" doesn't match the expected result for a server missing the module. See an example here: https://github.com/ansible/ansible/blob/devel/test/integration/targets/win_power_plan/tasks/main.yml

It also shows how to skip running the tests on server versions that don't support it.

Reply:
Yes, thank you! But it results in the same...

Reply:
I’ve done it in a few tests, have a look at the win_dsc ones as well.

Reply:
I have implemented it.